solr/core/src/java/org/apache/solr/schema/DenseVectorField.java

```diff
@@ -0,0 +1,200 @@
+        args.remove(KNN_VECTOR_DIMENSION);
+
+        this.similarityFunction = ofNullable(args.get(KNN_SIMILARITY_FUNCTION))
+                .map(value -> VectorSimilarityFunction.valueOf(value.toUpperCase(Locale.ROOT)))
+                .orElse(DEFAULT_SIMILARITY);
+        args.remove(KNN_SIMILARITY_FUNCTION);
+
+        this.properties &= ~MULTIVALUED;
+        this.properties &= ~UNINVERTIBLE;
+        
+        super.init(schema, args);
+    }
+
+    private int parseDimensionParameter(String value) {
+        try {
+            return Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, "the vector dimension must be an integer");
+        }
+    }
+
+    @Override
+    public void checkSchemaField(final SchemaField field) throws SolrException {
+        super.checkSchemaField(field);
+        if (field.multiValued()) {
+            throw new SolrException(SolrException.ErrorCode.SERVER_ERROR,
+                    getClass().getSimpleName() + " fields can not be multiValued: " + field.getName());
+        }
+    }
+    
+    public List<IndexableField> createFields(SchemaField field, Object value) {
+        List<IndexableField> fields = new ArrayList<>();
+        if (field.indexed()){
+            fields.add(createField(field, value));
+        }
+        if (field.stored()){
+            fields.add(createStoredField(field, value));
+        }
+        return fields;
+    }
+
+    private IndexableField createStoredField(SchemaField field, Object value) {
+        return new StoredField(field.getName(), value.toString());
+    }
+
+    @Override
+    public UninvertingReader.Type getUninversionType(SchemaField sf) {
+        return null;
+    }
+
+    @Override
+    public ValueSource getValueSource(SchemaField field, QParser parser) {
+        throw new SolrException(SolrException.ErrorCode.BAD_REQUEST,
+                "Function queries are not supported for Dense Vector fields.");
+    }
+
+    @Override
+    public void write(TextResponseWriter writer, String name, IndexableField f) throws IOException {
+        writer.writeVal(name, f.stringValue());
+    }
+
+    @Override
+    public IndexableField createField(SchemaField field, Object value) {
+        float[] parsedVector;
+        try {
+            parsedVector = parseVector(value.toString());
+        } catch (RuntimeException e) {
+            throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, "Error while creating field '" + field + "' from value '" + value + "', expected format:'[f1, f2, f3...fn]' e.g. [1.0, 3.4, 5.6]", e);
+        }
+        if (parsedVector == null) return null;
+
+        return new KnnVectorField(field.getName(), parsedVector, similarityFunction);
+    }
+
+    /**
+     * Parses a String vector.
+     *
+     * @param value with format: [f1, f2, f3, f4...fn]
```

Comment:
Have you considered an alternate more compressed string format? Maybe b64-encode the bytes of the float array?

Reply:
This should have been resolved with latest commit